File: covilo-web-ui/src/application/core/models/crime.ts
import { CrimeClassification } from "./crime-classification"
import { CrimePerpetrator } from "./crime-perpetrator"
import { City } from "./city"

/**
 *
 */
export interface Crime {
  date: Date
  city: City
  classification: CrimeClassification
  perpetrator?: CrimePerpetrator
  description: string
  confirmed: boolean
}

/**
 *
 */
export declare type Crimes = Crime[]

File: covilo-api/src/main/kotlin/me/kvdpxne/covilo/domain/models/Crime.kt
package me.kvdpxne.covilo.domain.models

import me.kvdpxne.covilo.utils.Recognizable
import java.io.Serializable
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.UUID

/**
 * @param identifier
 * @param date
 * @param city
 * @param classification
 * @param perpetrator
 * @param description
 * @param confirmed
 */
data class Crime(
  override val identifier: UUID = UUID.randomUUID(),
  var city: LocationCity,
  var classification: CrimeClassification,
  var perpetrator: CrimePerpetrator,
  var description: String,
  var confirmed: Boolean = false,
  override val createdDate: LocalDateTime = LocalDateTime.now(),
  override var lastModifiedDate: LocalDateTime? = null,
) : Recognizable, Auditable, Serializable {

  companion object {
    private const val serialVersionUID: Long = 1542620209L
  }
}

/**
 * Shortcut to the [Crime] collection.
 */
typealias Crimes = Collection<Crime>